src/interpreter/analytics.rs

```diff
@@ -0,0 +1,154 @@
+use std::collections::HashSet;
```

Comment:
I wouldn't put in in the `interpreter` module. For now, top-level module is fine.

Reply:
Or maybe let's stuff them in `interpreter_funcs/mod.rs`

Reply:
u sure? `interpreter_funcs/mod.rs` is quite nice and clean now, should we pollute it?